test/integration/multiple_deposit.test.ts

```diff
@@ -125,169 +143,793 @@ contract<ERC20RootVault, DeployOptions, CustomContext>(
+                                        .mul(24)
+                                        .mul(3600)
+                                        .mul(365)
+                                );
+
+                        // expected management fee = first fee + second fee
+                        let expectedManagementFee =
+                            expectedManagementFeeFirst.add(
+                                expectedManagementFeeSecond
+                            );
+
+                        let managementFeeAbsDifference = expectedManagementFee
+                            .sub(realManagementFee)
+                            .abs();
+
+                        /* 
+                            MANAGEMENT FEES
+                            dif <= 0.01 * 1% * fee
+                            dif * 100 * 100 <= fee
+                            dif * (10 ** 4) - fee <= 0
+                        */
+                        expect(
+                            managementFeeAbsDifference
+                                .mul(10000)
+                                .sub(realManagementFee)
+                                .lte(0)
+                        ).to.be.true;
+
+                        // PERFORMANCE FEES
+                        expect(expectdPerformanceFee).to.be.equal(
+                            realPerformanceFee
+                        );
+                    }
+
+                    /* 
+                        --------------------- COLLECT ALL FEES ---------------------------
+                    */
+
+                    // collect management fees
+                    if (
+                        (await this.subject.balanceOf(strategyTreasury)).gt(0)
+                    ) {
+                        await this.subject.withdraw(
+                            strategyTreasury,
+                            BigNumber.from(2).pow(256).sub(1),
+                            [0, 0]
+                        );
+                    }
+
+                    // collect performance fees
+                    if (
+                        (
+                            await this.subject.balanceOf(
+                                strategyPerformanceTreasury
+                            )
+                        ).gt(0)
+                    ) {
+                        await this.subject.withdraw(
+                            strategyPerformanceTreasury,
+                            BigNumber.from(2).pow(256).sub(1),
+                            [0, 0]
+                        );
+                    }
+
+                    // collect protocol fees
+                    if (
+                        (await this.subject.balanceOf(protocolTreasury)).gt(0)
+                    ) {
+                        await this.subject.withdraw(
+                            protocolTreasury,
+                            BigNumber.from(2).pow(256).sub(1),
+                            [0, 0]
+                        );
```

Comment:
Can we also add a post-withdraw check in terms of real tokens? E.g. expect ETH == ETH

Reply:
I already have these post-withdraw checks for WETH and USDC in the end if the test